src/libraries/System.Private.CoreLib/src/System/Globalization/TimeSpanFormat.cs

```diff
@@ -137,24 +144,33 @@ private static bool TryFormatStandard(TimeSpan value, StandardFormat format, str
                     // "c": Write out a fraction only if it's non-zero, and write out all 7 digits of it.
                     if (fraction != 0)
                     {
+                        Debug.Assert(decimalSeparator is null);
                         fractionDigits = DateTimeFormat.MaxSecondsFractionDigits;
                         requiredOutputLength += fractionDigits + 1; // digits plus leading decimal separator
                     }
                     break;
 
                 case StandardFormat.G:
                     // "G": Write out a fraction regardless of whether it's 0, and write out all 7 digits of it.
+                    Debug.Assert(decimalSeparator is not null);
                     fractionDigits = DateTimeFormat.MaxSecondsFractionDigits;
-                    requiredOutputLength += fractionDigits + 1; // digits plus leading decimal separator
+                    requiredOutputLength += fractionDigits;
+                    requiredOutputLength += typeof(TChar) == typeof(char) ?
+                        decimalSeparator.Length :
+                        Encoding.UTF8.GetByteCount(decimalSeparator);
```

Comment:
>Encoding.UTF8.GetByteCount(decimalSeparator); [](http://example.com/codeflow?start=24&length=45)

maybe better initialize the length when initializing decimalSeparator and avoid calling this every time we format?